src/MongoDB.Driver/LibMongoCrypt/LibMongoCryptController.cs

```diff
@@ -0,0 +1,415 @@
+        {
+            var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            socket.Connect(request.Endpoint, 443);
+
+            var validationCallback = new RemoteCertificateValidationCallback(AcceptAnyCertificate); // TODO: what validation needs to be done?
+
+            using (var networkStream = new NetworkStream(socket, ownsSocket: true))
+            using (var sslStream = new SslStream(networkStream, leaveInnerStreamOpen: false, validationCallback))
+            {
+#if NETSTANDARD1_5
+                sslStream.AuthenticateAsClientAsync(request.Endpoint).ConfigureAwait(false).GetAwaiter().GetResult();
+#else
+                sslStream.AuthenticateAsClient(request.Endpoint);
+#endif
+
+                var requestBytes = request.Message.ToArray();
+                sslStream.Write(requestBytes);
+
+                var buffer = new byte[4096];
+                while (request.BytesNeeded > 0)
+                {
+                    var count = sslStream.Read(buffer, 0, buffer.Length);
+                    var responseBytes = new byte[count];
+                    Buffer.BlockCopy(buffer, 0, responseBytes, 0, count);
+                    request.Feed(responseBytes);
+                }
+            }
+        }
+
+        private async Task SendKmsRequestAsync(KmsRequest request, CancellationToken cancellation)
+        {
+            var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+#if NETSTANDARD1_5
+            await socket.ConnectAsync(request.Endpoint, 443).ConfigureAwait(false);
+#else
+            await Task.Factory.FromAsync(socket.BeginConnect(request.Endpoint, 443, null, null), socket.EndConnect).ConfigureAwait(false);
+#endif
+
+            var validationCallback = new RemoteCertificateValidationCallback(AcceptAnyCertificate); // TODO: what validation needs to be done?
+
+            using (var networkStream = new NetworkStream(socket, ownsSocket: true))
+            using (var sslStream = new SslStream(networkStream, leaveInnerStreamOpen: false, validationCallback))
+            {
+                await sslStream.AuthenticateAsClientAsync(request.Endpoint).ConfigureAwait(false);
+
+                var requestBytes = request.Message.ToArray();
+                await sslStream.WriteAsync(requestBytes, 0, requestBytes.Length).ConfigureAwait(false);
+
+                var buffer = new byte[4096];
+                while (request.BytesNeeded > 0)
+                {
+                    var count = await sslStream.ReadAsync(buffer, 0, buffer.Length).ConfigureAwait(false);
+                    var responseBytes = new byte[count];
+                    Buffer.BlockCopy(buffer, 0, responseBytes, 0, count);
+                    request.Feed(responseBytes);
+                }
+            }
+        }
+
+        private void RestoreDbNodeInResponse(BsonDocument request, BsonDocument response)
+        {
```

Comment:
`MongoCryptD` lose `$db` node in the response. So, in case if this value was presented in the request, we restore it